### ISSUE TITLE
How to convert a TF1 model with custom_getter to TF2

### ISSUE THREAD

**Issue by emadboctorx** [tensorflow/tensorflow#47342]
I need to do this very same calculation in TF2 which ends up by calculating `v1` and `v2` which are softmax output and the same softmax output but calculated using a 
`tf.train.ExponentialMovingAverage` `average()` method. The original example can be found [here](https://github.com/openai/baselines/blob/master/baselines/acer/acer.py) at line 88 but I simplified the code in the example here for readability.

**TF1 code:**

    import tensorflow as tf
    
    
    with tf.variable_scope('my_model', reuse=tf.AUTO_REUSE):
        step_model = StepModel(step_param)  # step_param is a placeholder
        train_model = TrainModel(train_param)  # train_param is a placeholder
    
    params = tf.trainable_variables('my_model')
    ema = tf.train.ExponentialMovingAverage(0.99)
    ema_apply_op = ema.apply(params)
    
    def custom_getter(getter, *args, **kwargs):
        return ema.average(getter(*args, **kwargs))
    
    with tf.variable_scope("my_model", custom_getter=custom_getter, reuse=True):
        avg_model = AveragingModel(avg_param)  # avg_param is a placeholder
    
    train_model_p = tf.nn.softmax(train_model.pi)
    avg_model_p = tf.nn.softmax(avg_model.pi)
    common_param = some_param
    with tf.get_default_session() as sess:
        feed_dict = {
            train_model.train_param: common_param,
            avg_model.avg_param: common_param,
        }
        ops = [train_model_p, avg_model_p]
        v1, v2 = sess.run(ops, feed_dict)[1:]

If I want to keep a moving average of the trainable variables in TF2 without worrying about `v1` and `v2`, I'd do:

    from tensorflow.keras.optimizers import Adam
    from tensorflow_addons.optimizers import MovingAverage
    from tensorflow.keras.models import Model

and then I'd wrap the `tf.keras.optimizers.Optimizer()` being used:

    
    model = Model(...)
    optim = MovingAverage(Adam())
    model.compile(optimizer=optim)
    model.fit(...)

But I need to get the values of `v1` and `v2` for further calculations.

**Comment by amahendrakar** [comment]
@emadboctorx,
Please go through [this guide](https://www.tensorflow.org/guide/migrate) to migrate your code from TF 1.x to TF 2.x.

Also, this question is better asked on [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow) since it is not a TensorFlow bug or feature request. There is also a larger community that reads questions there. Thanks!
